### PR TITLE
Fix `ListColumn.to_pandas()` to retain `list` type

### PR DESCRIPTION
**Pull request by galipremsagar**
## Description
Fixes: #14568 

This PR fixes `ListColumn.to_pandas()` by calling `ArrowArray.to_pylist()` method to retain `list` type in pandas series.
 
## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/core/column/lists.py [hunk 1]
```diff
@@ -288,6 +289,23 @@ def _transform_leaves(self, func, *args, **kwargs) -> Self:
...
         return lc
 
+    def to_pandas(
+        self,
+        *,
+        index: Optional[pd.Index] = None,
+        nullable: bool = False,
+    ) -> pd.Series:
+        # Can't rely on Column.to_pandas implementation for lists.
+        # Need to perform `to_pylist` to preserve list types.
```
Comment (vyasr) [inline python/cudf/cudf/core/column/lists.py:299]: Presumably this is some bug in how list dtypes are handled in arrow? Do we need to report something upstream?

**Review by vyasr — APPROVED** [review]
Approved, but with one potential follow-up request.

File: python/cudf/cudf/core/column/lists.py [hunk 2]
```diff
@@ -288,6 +289,23 @@ def _transform_leaves(self, func, *args, **kwargs) -> Self:
...
+        if nullable:
+            raise NotImplementedError(f"{nullable=} is not implemented.")
+
+        pd_series = pd.Series(self.to_arrow().to_pylist())
```
Comment (mroeschke) [inline python/cudf/cudf/core/column/lists.py:303]:
```suggestion
        pd_series = pd.Series(self.to_arrow().to_pylist(), dtype="object")
```

(Just to be safe)

**Review by mroeschke — APPROVED** [review]

File: python/cudf/cudf/core/column/lists.py (hunk 1 above)
Comment (galipremsagar) [reply python/cudf/cudf/core/column/lists.py:299]: There is already an open issue: https://github.com/apache/arrow/issues/34574

File: python/dask_cudf/dask_cudf/tests/test_groupby.py [hunk 3]
```diff
@@ -702,13 +702,9 @@ def test_is_supported(arg, supported):
 
 def test_groupby_unique_lists():
     df = pd.DataFrame({"a": [0, 0, 0, 1, 1, 1], "b": [10, 10, 10, 7, 8, 9]})
-    ddf = dd.from_pandas(df, 2)
     gdf = cudf.from_pandas(df)
     gddf = dask_cudf.from_cudf(gdf, 2)
-    dd.assert_eq(
-        ddf.groupby("a").b.unique().compute(),
-        gddf.groupby("a").b.unique().compute(),
-    )
```
Comment (rjzamora) [inline python/dask_cudf/dask_cudf/tests/test_groupby.py:711]: Does this no longer pass?

Comment (galipremsagar) [reply python/dask_cudf/dask_cudf/tests/test_groupby.py:711]: Yes, we fixed a bug that `dask` is yet to fix:
```
In [6]: df = pd.DataFrame({"a": [0, 0, 0, 1, 1, 1], "b": [10, 10, 10, 7, 8, 9]})

In [7]: df.groupby("a").b.unique()
Out[7]: 
a
0         [10]
1    [7, 8, 9]
Name: b, dtype: object

In [8]: import dask.dataframe as dd

In [9]: ddf = dd.from_pandas(df, 2)

In [10]: ddf.groupby("a").b.unique().compute()
Out[10]: 
a
0       [10]
1    [7 8 9]
Name: b, dtype: string

In [11]: ddf.groupby("a").b.unique().compute()[1]
Out[11]: '[7 8 9]' # Wrong, Should be a list of 3 numbers `[7, 8, 9]` than a raw string.
```

Comment (rjzamora) [reply python/dask_cudf/dask_cudf/tests/test_groupby.py:711]: Ah okay, thanks for the info. Looks like it works with the dask-expr backend, but this change makes sense to me. Thanks!

**Review by rjzamora — APPROVED** [review]

**Pull request merged**